python-ecosys/urequests/urequests.py

```diff
@@ -130,3 +190,15 @@ def patch(url, **kw):
 
 def delete(url, **kw):
     return request("DELETE", url, **kw)
+
+
+def encode_basic_auth(username, password):
```

Comment:
Does this need to be a separate function?  It would be more efficient and smaller bytecode if it were inlined at its point of use.

Reply:
Minimising the overhead of extra functions is a good goal. I've done this here as well as the is_chunked_data() one.